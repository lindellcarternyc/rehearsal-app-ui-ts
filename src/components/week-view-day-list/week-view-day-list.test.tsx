import * as React from 'react'
import * as renderer from 'react-test-renderer'

import WeekViewDayList, { WeekViewDayListProps } from './week-view-day-list'

import WEEKS from '../../mocks/data/weeks'
const selectDay = jest.fn()
const showAddRehearsal = jest.fn()

describe('WeekViewDayList', () => {
  it('renders without rehearsals', () => {
    const props: WeekViewDayListProps = {
      week: WEEKS[0],
      selectDay,
      showAddRehearsal
    }
    const list = renderer.create(<WeekViewDayList {...props} />)
    expect(list).toMatchSnapshot()
  })

  it('renders with rehearsal', () => {
    const props: WeekViewDayListProps = {
      week: WEEKS[1],
      selectDay,
      showAddRehearsal
    }
    const list = renderer.create(<WeekViewDayList {...props} />)
    expect(list).toMatchSnapshot()
  })
})