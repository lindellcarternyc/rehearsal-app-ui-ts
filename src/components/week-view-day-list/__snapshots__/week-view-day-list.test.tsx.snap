// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekViewDayList renders with rehearsal 1`] = `
<div
  className="ui divided selection relaxed list"
  role="list"
>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Sun, January 14
    </div>
    <p
      style={
        Object {
          "paddingLeft": "1rem",
        }
      }
    >
      No Rehearsals Today
    </p>
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Mon, January 15
    </div>
    <p
      style={
        Object {
          "paddingLeft": "1rem",
        }
      }
    >
      No Rehearsals Today
    </p>
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Tue, January 16
    </div>
    <div
      className="ui bulleted list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        19:00 - 22:00
      </div>
    </div>
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Wed, January 17
    </div>
    <p
      style={
        Object {
          "paddingLeft": "1rem",
        }
      }
    >
      No Rehearsals Today
    </p>
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Thu, January 18
    </div>
    <p
      style={
        Object {
          "paddingLeft": "1rem",
        }
      }
    >
      No Rehearsals Today
    </p>
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Fri, January 19
    </div>
    <p
      style={
        Object {
          "paddingLeft": "1rem",
        }
      }
    >
      No Rehearsals Today
    </p>
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="header"
    >
      Sat, January 20
    </div>
    <div
      className="ui bulleted list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        13:00 - 14:30
      </div>
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        14:30 - 16:00
      </div>
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        16:00 - 18:00
      </div>
    </div>
  </div>
</div>
`;

exports[`WeekViewDayList renders without rehearsals 1`] = `
<div>
  <h3
    className="ui header"
  >
    No Rehearsals Sun, January 7 - Sat, January 13
  </h3>
  <button
    className="ui button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    Add Rehearsal
  </button>
</div>
`;
